File: src/RendevumVar.Core/Entities/Review.cs
namespace RendevumVar.Core.Entities;

public class Review : BaseEntity
{
    public Guid AppointmentId { get; set; }
    public Guid CustomerId { get; set; }
    public Guid SalonId { get; set; }
    public Guid? StaffId { get; set; }
    public int Rating { get; set; } // 1-5
    public string? Comment { get; set; }
    public string? Response { get; set; }
    public Guid? ResponseBy { get; set; }
    public DateTime? ResponseAt { get; set; }
    public bool IsPublished { get; set; } = true;

    // Navigation properties
    public Appointment Appointment { get; set; } = null!;
    public User Customer { get; set; } = null!;
    public Salon Salon { get; set; } = null!;
    public Staff? Staff { get; set; }
    public User? ResponseByUser { get; set; }
}

File: frontend/src/types/review.ts
export interface Review {
  id: string;
  appointmentId: string;
  customerId: string;
  customerName: string;
  salonId: string;
  salonName: string;
  staffId?: string;
  staffName?: string;
  rating: number;
  comment?: string;
  response?: string;
  responseBy?: string;
  responseByName?: string;
  responseAt?: string;
  isPublished: boolean;
  createdAt: string;
}

export interface CreateReviewDto {
  appointmentId: string;
  salonId: string;
  staffId?: string;
  rating: number;
  comment?: string;
}

export interface UpdateReviewDto {
  rating: number;
  comment?: string;
}

export interface ReviewResponseDto {
  response: string;
}

export interface SalonRating {
  salonId: string;
  averageRating: number;
  totalReviews: number;
  ratingDistribution: {
    [key: number]: number;
  };
}

export interface StaffRating {
  staffId: string;
  averageRating: number;
  totalReviews: number;
}
